klippy/extras/probe.py

```diff
@@ -154,16 +156,24 @@ def run_probe(self, gcmd):
                                            self.samples_tolerance, minval=0.)
         samples_retries = gcmd.get_int("SAMPLES_TOLERANCE_RETRIES",
                                        self.samples_retries, minval=0)
+        samples_discard_first = gcmd.get_int("SAMPLES_DISCARD_FIRST",
+                                             self.samples_discard, minval=0)
         samples_result = gcmd.get("SAMPLES_RESULT", self.samples_result)
         must_notify_multi_probe = not self.multi_probe_pending
         if must_notify_multi_probe:
             self.multi_probe_begin()
         probexy = self.printer.lookup_object('toolhead').get_position()[:2]
         retries = 0
+        discards = 0
         positions = []
         while len(positions) < sample_count:
             # Probe position
             pos = self._probe(speed)
+            # Discard the first readings (probably at most one)
+            if discards <= samples_discard_first:
+                gcmd.respond_info("Probe sample discarded")
+                discards += 1
+                continue
```

Comment:
This continue will skip the retract down the loop that must happen to avoid "Probe triggered prior to movement" error

Reply:
Thanks, I believe I've fixed that.